geode-core/src/test/java/org/apache/geode/distributed/AbstractLauncherTest.java

```diff
@@ -14,253 +14,370 @@
-    assertFalse(AbstractLauncher.isSet(properties, NAME));
+  @Test
+  public void isSetReturnsFalseIfPropertyHasEmptyValue() throws Exception {
+    Properties properties = new Properties();
 
     properties.setProperty(NAME, "");
 
-    assertTrue(properties.containsKey(NAME));
-    assertFalse(AbstractLauncher.isSet(properties, NAME));
+    assertThat(AbstractLauncher.isSet(properties, NAME)).isFalse();
+  }
+
+  @Test
+  public void isSetReturnsFalseIfPropertyHasBlankValue() throws Exception {
+    Properties properties = new Properties();
 
     properties.setProperty(NAME, "  ");
 
-    assertTrue(properties.containsKey(NAME));
-    assertFalse(AbstractLauncher.isSet(properties, NAME));
+    assertThat(AbstractLauncher.isSet(properties, NAME)).isFalse();
+  }
+
+  @Test
+  public void isSetReturnsFalseIfPropertyHasRealValue() throws Exception {
+    Properties properties = new Properties();
 
     properties.setProperty(NAME, "memberOne");
 
-    assertTrue(AbstractLauncher.isSet(properties, NAME));
-    assertFalse(AbstractLauncher.isSet(properties, "NaMe"));
+    assertThat(AbstractLauncher.isSet(properties, NAME)).isTrue();
   }
 
   @Test
-  public void testLoadGemFirePropertiesWithNullURL() {
-    final Properties properties = AbstractLauncher.loadGemFireProperties(null);
-    assertNotNull(properties);
-    assertTrue(properties.isEmpty());
+  public void isSetKeyIsCaseSensitive() throws Exception {
+    Properties properties = new Properties();
+
+    properties.setProperty(NAME, "memberOne");
+
+    assertThat(AbstractLauncher.isSet(properties, "NaMe")).isFalse();
   }
 
   @Test
-  public void testLoadGemFirePropertiesWithNonExistingURL() throws MalformedURLException {
-    final Properties properties = AbstractLauncher
-        .loadGemFireProperties(new URL("file:///path/to/non_existing/gemfire.properties"));
-    assertNotNull(properties);
-    assertTrue(properties.isEmpty());
+  public void loadGemFirePropertiesWithNullURLReturnsEmptyProperties() throws Exception {
+    URL nullUrl = null;
+
+    Properties properties = AbstractLauncher.loadGemFireProperties(nullUrl);
+
+    assertThat(properties).isNotNull().isEmpty();
   }
 
   @Test
-  public void testGetDistributedSystemProperties() {
-    AbstractLauncher<?> launcher = createAbstractLauncher("memberOne", "1");
+  public void loadGemFirePropertiesWithNonExistingURLReturnsEmptyProperties() throws Exception {
+    URL nonExistingUrl = new URL("file:///path/to/non_existing/gemfire.properties");
+
+    Properties properties = AbstractLauncher.loadGemFireProperties(nonExistingUrl);
 
-    assertNotNull(launcher);
-    assertEquals("1", launcher.getMemberId());
-    assertEquals("memberOne", launcher.getMemberName());
+    assertThat(properties).isNotNull().isEmpty();
+  }
 
-    Properties distributedSystemProperties = launcher.getDistributedSystemProperties();
+  @Test
+  public void abstractLauncherHasMemberIdAndMemberName() throws Exception {
```

Comment:
I don't see the need for all the tests that exercise methods that are overridden in the embedded test class for AbstractLauncher. Any test that calls, for instance, 'launcher.getMemberId()' is only testing the test class, not the default implementation in the abstract class.